rundetection/rules/loq_rules.py

```diff
@@ -0,0 +1,192 @@
+"""
+Rules for LOQ
+"""
+
+from __future__ import annotations
+
+import logging
+import typing
+from dataclasses import dataclass
+from pathlib import Path
+
+import requests
+import xmltodict
+
+from rundetection.rules.rule import Rule
+
+if typing.TYPE_CHECKING:
+    from rundetection.job_requests import JobRequest
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class SansFileData:
+    title: str
+    type: str
+    run_number: str
+
+
+def _extract_run_number_from_filename(filename: str) -> str:
+    # Assume filename looks like so: LOQ00100002.nxs
+    # Return everything between 1 and 2 including 1 and 2.
+    filename = filename.split(".")[0]  # Drop .nxs
+    filename = filename.strip("LOQ")  # Drop LOQ
+
+    # Drop first 0s
+    run_number = ""
+    first_non_zero_found = False
+    for char in filename:
+        if first_non_zero_found is False and char != "0":
+            run_number += char
+            first_non_zero_found = True
+        elif first_non_zero_found:
+            run_number += char
+    return run_number
```

Comment:
Bit of a confusing code block. Why not use `lstrip()`?
```suggestion
    return filename.split(".")[0]lstrip("LOQ").lstrip("0")
```